docs/Toy/toy/tests/rewrites/test_dce.py

```diff
@@ -0,0 +1,24 @@
+from xdsl.builder import Builder
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.pattern_rewriter import PatternRewriteWalker
+from ...rewrites.dead_code_elimination import RemoveUnusedOperations
+from ...dialects import toy
+
+
+def test_dce():
+    @ModuleOp
+    @Builder.implicit_region
+    def module():
+        a = toy.ConstantOp.from_list([1, 2, 3], [3]).res
+        _b = toy.ConstantOp.from_list([1, 2, 3], [3]).res
+        toy.PrintOp(a)
+
+    @ModuleOp
+    @Builder.implicit_region
+    def expected():
+        a = toy.ConstantOp.from_list([1, 2, 3], [3]).res
+        toy.PrintOp(a)
+
+    PatternRewriteWalker(RemoveUnusedOperations()).rewrite_module(module)
```

Comment:
Would it make sense to have a `dce()` function defined in dead_code_elimination that takes a module as an input and contains this call to the PatternRewriter?

Reply:
I agree with this. Constructing the walker and so on should be in the other file, IMO.